src/main/groovy/com/calgaryscientific/gradle/VeracodeBeginScanTask.groovy

```diff
@@ -26,28 +26,78 @@
 
 package com.calgaryscientific.gradle
 
+import groovy.transform.CompileStatic
+
+@CompileStatic
 class VeracodeBeginScanTask extends VeracodeTask {
     static final String NAME = 'veracodeBeginScan'
+    private String app_id
+    private Set<String> moduleWhitelist
 
     VeracodeBeginScanTask() {
-        description = 'Starts a Veracode scan for the application id passed in'
+        description = 'Starts a Veracode scan for given application ID'
         requiredArguments << 'app_id'
+        dependsOn "veracodeGetPreScanResults"
+        app_id = project.findProperty("app_id")
+        defaultOutputFile = new File("${project.buildDir}/veracode", 'begin-scan.xml')
     }
 
-    void run() {
-        def moduleIds = []
-        def whiteList = readListFromFile(new File("src/apps/${project.app_id}/modules-whitelist.txt"))
-        readXml('build/pre-scan-results.xml').each() { module ->
-            if (whiteList.contains(module.@name)) {
-                moduleIds << module.@id
+    VeracodeGetPreScanResultsTask preScan = new VeracodeGetPreScanResultsTask()
+    File preScanResultsOutputFile = preScan.getOutputFile()
+
+    static Set<String> extractWhitelistModuleIds(Node xml, Set<String> whitelist) {
+        Set<String> moduleIds = []
+        Set<String> moduleNames = []
+        NodeList moduleList = xml.getAt("module") as NodeList
+        for (int i = 0; i < moduleList.size(); i++) {
+            Node moduleEntry = moduleList.get(i) as Node
+            String id = moduleEntry.attribute('id')
+            String name = moduleEntry.attribute('name')
+            String status = moduleEntry.attribute('status')
+            if (!status.startsWith('(Fatal)') && whitelist.contains(name)) {
+                moduleIds << id
+                moduleNames << name
+                printf "Selecting module: %s - %s\n", name, status
             }
         }
-        println "Modules in whitelist: ${whiteList.size()}"
-        println "Modules selected: ${moduleIds.size()}"
-        if (whiteList.size() != moduleIds.size()) {
-            println 'WARNING: Not all the files in whitelist are being scanned. Some modules no longer exist? Manual whitelist maintenance should be performed.'
+        if (whitelist.size() != moduleNames.size()) {
```

Comment:
what does this if statement do?

Reply:
Checks if the whitelist is bigger than the whitelist modules found. If that is the case then it "warns" that not all whitelist modules where found.

Reply:
I see ... so you might want to extract this to something like warnIfNotAllWhitelistModulesFound?